app/src/main/java/com/example/films/MoviesListAdapter.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.films
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.RatingBar
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+
+class MoviesListAdapter(
+    context: Context,
+    var movies : List<Movie>,
+    private val clickListener: MoviesListClickListener?
+) : RecyclerView.Adapter<MoviesListViewHolder>() {
+
+    private val inflater : LayoutInflater = LayoutInflater.from(context)
+
+    override fun getItemCount() = movies.size
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MoviesListViewHolder {
+        return MoviesListViewHolder(inflater.inflate(R.layout.view_holder_movie, parent, false))
+    }
+
+    override fun onBindViewHolder(holder: MoviesListViewHolder, position: Int) {
+        holder.bind(movies[position])
+        holder.itemView.setOnClickListener {
```

Comment:
create click listener every time when viewholder binded it's not good idea
let's set click listener when your viewholder created

Reply:
How can I do it? Do I need to store movie in viewHolder (as one of fields) ? Because there is not anything about current movie in "override fun onCreateViewHolder(parent: ViewGroup, viewType: Int)", and for movieOnClick method I need to know movie
Now viewHolder gets some movie only in function bind(movie)

Reply:
yeap you need store movie as global value